src/node/share_manager.h

```diff
@@ -153,44 +146,65 @@ namespace ccf
       return encrypted_shares;
     }
 
+    void _shuffle_recovery_shares(
+      kv::Tx& tx, const LedgerSecret& latest_ledger_secret)
+    {
+      auto ls_wrapping_key = LedgerSecretWrappingKey();
+      auto wrapped_latest_ls = ls_wrapping_key.wrap(latest_ledger_secret);
+      auto recovery_shares = tx.rw(network.shares);
+      recovery_shares->put(
+        0, {wrapped_latest_ls, compute_encrypted_shares(tx, ls_wrapping_key)});
+    }
+
     void set_recovery_shares_info(
       kv::Tx& tx,
       const LedgerSecret& latest_ledger_secret,
-      const std::optional<LedgerSecret>& previous_ledger_secret = std::nullopt,
-      kv::Version latest_ls_version = kv::NoVersion)
+      const std::optional<VersionedLedgerSecret>& previous_ledger_secret =
+        std::nullopt,
+      std::optional<kv::Version> latest_ls_version = std::nullopt)
     {
       // First, generate a fresh ledger secrets wrapping key and wrap the
       // latest ledger secret with it. Then, encrypt the penultimate ledger
       // secret with the latest ledger secret and split the ledger secret
-      // wrapping key, allocating a new share for each active member. Finally,
-      // encrypt each share with the public key of each member and record it in
-      // the shares table.
+      // wrapping key, allocating a new share for each active recovery member.
+      // Finally, encrypt each share with the public key of each member and
+      // record it in the shares table.
 
-      auto ls_wrapping_key = LedgerSecretWrappingKey();
-      auto wrapped_latest_ls = ls_wrapping_key.wrap(latest_ledger_secret);
+      _shuffle_recovery_shares(tx, latest_ledger_secret);
```

Comment:
Technically this is reserved, and we don't use leading underscore anywhere else in the code base that I know of, I would prefer not to introduce it unless it's a deliberate and well-defined convention.